launcher/ui/dialogs/CopyInstanceDialog.cpp

```diff
@@ -117,6 +122,40 @@ QString CopyInstanceDialog::instGroup() const
     return ui->groupBox->currentText();
 }
 
+const InstanceCopyPrefs& CopyInstanceDialog::getChosenOptions() const
+{
+    return m_selectedOptions;
+}
+
+void CopyInstanceDialog::checkAllCheckboxes(const bool& b)
+{
+    ui->keepPlaytimeCheckbox->setChecked(b);
+    ui->copySavesCheckbox->setChecked(b);
+    ui->copyGameOptionsCheckbox->setChecked(b);
+    ui->copyResPacksCheckbox->setChecked(b);
+    ui->copyShaderPacksCheckbox->setChecked(b);
+    ui->copyServersCheckbox->setChecked(b);
+    ui->copyModsCheckbox->setChecked(b);
+}
+
+// Sets b to true if state is a checked checkbox
+void CopyInstanceDialog::checkBool(bool& b, const int& state)
```

Comment:
```suggestion
bool CopyInstanceDialog::checkBool(int state)
```

also change the header!

Reply:
I wrote the header this way because I'm updating the bool inside, then checking if everything in m_selectedOptions is true, and finally checking the "Select all" check box accordingly. I thought it'd be better to do this all within the same function since we need to update the Select all checkbox every time we change a value in m_selectOptions anyways, so we'd save some code.

Currently:
```C++
checkBool(m_selectedOptions.copySaves, state); // Update copySaves and select all checkbox
```

If I change it to return a new boolean instead, I would need to update the select all checkbox separately once the value has changed in m_selectOptions (unless there's a simpler way to do this) so It would be something like:

```C++
m_selectedOptions.copySaves = checkBool(state); // Update copySaves
updateSelectAllCheckbox(); // <-- new function
```

Is the new one prefereable? (So I know in the future)

Reply:
i think it's best to separate those two. messing with argument references tends to be harder to read and more error-prone then simply returning a value, and then assigning it to whatever variable you want.

I'd also change `checkBool` to a more descriptive name now :p

Reply:
Okay! And yeah a function is not really needed, also should the updateSelectAllCheckbox() function be in the private slots area of the CopyInstanceDialog class? Or should it be in the regular private section with the data?

Reply:
doesn't really matter. since you're not connecting any signals to it, a non-slot works just fine

Reply:
Done! commit 385c452